index.js

```diff
@@ -1,42 +1,51 @@
-import React, { PureComponent } from 'react'
-
-class GitHubButton extends PureComponent {
-  constructor (props) {
-    super(props)
-    this.$ = React.createRef()
-    this._ = React.createRef()
-  }
-  render () {
-    return React.createElement('span', { ref: this.$ }, React.createElement('a', { ...this.props, ref: this._ }, this.props.children))
-  }
-  componentDidMount () {
-    this.paint()
-  }
-  getSnapshotBeforeUpdate () {
-    this.reset()
-    return null
-  }
-  componentDidUpdate () {
-    this.paint()
-  }
-  componentWillUnmount () {
-    this.reset()
-  }
-  paint () {
-    const _ = this.$.current.appendChild(document.createElement('span'))
-    import(/* webpackMode: "eager" */ 'github-buttons').then(({ render }) => {
-      if (this._.current != null) {
-        render(_.appendChild(this._.current), function (el) {
-          try {
-            _.parentNode.replaceChild(el, _)
-          } catch (_) {}
-        })
+import React, { useCallback, useEffect, useRef } from "react";
+
+const GitHubButton = React.memo(({ children, ...props }) => {
+  const containerRef = useRef(null);
+  const buttonRef = useRef(null);
+
+  const paint = useCallback(() => {
+    if (!containerRef.current) return;
+
+    const tempSpan = document.createElement("span");
+    containerRef.current.appendChild(tempSpan);
+
+    import(/* webpackMode: "eager" */ "github-buttons")
+      .then(({ render }) => {
+        if (buttonRef.current && containerRef.current) {
```

Comment:
Also check if `tempSpan` is `lastChild` here. - It's "possible" that the dynamic `import()` is too slow that react has triggered rendering of the component more than once, when the library is still loading. By checking if `tempSpan` is still the `lastChild`, we can ensure only the "latest" rendering will actually compute the button.

Reply:
Basically the logic should be somewhat similar to the vue implementation, that this check should run twice for two async gaps to avoid rendering with previous props before update.
https://github.com/buttons/vue-github-button/blob/091412bdc4c2a88a93825721f1cbbc97a8229146/index.js#L47-L49
https://github.com/buttons/vue-github-button/blob/091412bdc4c2a88a93825721f1cbbc97a8229146/index.js#L51-L53